tests/lax_numpy_reducers_test.py

```diff
@@ -770,5 +770,72 @@ def test_f16_mean(self, dtype):
     self.assertAllClose(expected, actual, atol=0)
 
 
+
+  def _is_canonical(dtype):
+    _dtype = dtypes.dtype(dtype)
+    return _dtype == dtypes.canonicalize_dtype(_dtype)
+
+  @jtu.sample_product(
+    [dict(shape=shape, axis=axis)
+      for shape in all_shapes
+      for axis in list(
+        range(-len(shape), len(shape))
+      ) + ([None] if len(shape) == 1 else [])],
+    dtype=filter(_is_canonical, all_dtypes),
+    out_dtype=filter(_is_canonical, all_dtypes),
```

Comment:
I think the filter can be removed: I believe `all_dtypes` is defined in a way that makes this unnecessary.